markdown.code-snippets

```diff
@@ -44,7 +44,26 @@
 			""
 		],
 		"description": "Inserts dynamic index tag"
+	},
+	"first-paragraph tag": {
+		"scope": "markdown",
+		"prefix": [
+			"fst"
+		],
+		"body": [
+			"{:.first}$0"
```

Comment:
Does this one need a newline, like the sidenote one has below?

Reply:
@LouiseSteward Good question. As you may know, in kramdown you can put the tag above or below the paragraph – we've historically put tags after paragraphs, but this is really by accident. In the Aaron Reed project, I've been putting some tags, like this one, above paragraphs rather than below, where that makes the logic of the tagging more readable/understandable. 

I think there's some kind of difference between tags that tag for *purpose* (like `first`, `box`, or `sidenote`), and tags that are for layout (e.g. refinement `tighten` tags). 'Purpose' tags are probably better placed before the element they tag, so that when reading the markdown you know you're now reading, say, a  `first` paragraph, or a `box` or `sidenote`. In those cases, we don't want the snippet adding a line break.